File: LexerPLSharpVersion/LexerPLSharpVersion/constants/WhiteChars.cs
using LexerPLSharpVersion.token;

namespace LexerPLSharpVersion.constants {
   public static class WhiteChars {
      private const char WhiteSpace = ' ';
      private const char NewLine = '\n';
      private const char Tabulator = '\t';
      private const char CarriageReturn = '\r';
      private const string EscapedCarriageReturn = "\\r";
      private const string EscapedTabulator = "\\t";
      public const string EscapedNewLine = "\\n";

      public static bool IsWhiteChar(char character) {
         switch (character){
            case NewLine:
            case Tabulator:
            case CarriageReturn:
            case WhiteSpace:
               return true;
            default:
               return false;
         }
      }

      public static TokenType DetectWhiteCharType(char character) {
         switch (character){
            case WhiteSpace:
               return TokenType.WhiteSpace;
            case Tabulator:
               return TokenType.Tabulator;
            case CarriageReturn:
               return TokenType.CarriageReturn;
            case NewLine:
               return TokenType.NewLine;
            default:
               return TokenType.Unknown;
         }
      }
      
      public static string EscapeWhiteChars(char character) {
         switch (character){
            case NewLine:
               return EscapedNewLine;
            case Tabulator:
               return EscapedTabulator;
            case CarriageReturn:
               return EscapedCarriageReturn;
            case WhiteSpace:
               return " ";
         }

         return "";
      }
   }
}

File: ITEJA_Yarosh/src/main/java/lexer/constants/WhiteChars.java
package lexer.constants;

import token.TokenType;

public class WhiteChars {
    public static final char WHITE_SPACE = ' ';
    public static final char NEW_LINE = '\n';
    public static final char TABULATOR = '\t';
    public static final char CARRIAGE_RETURN = '\r';
    public static final String ESCAPED_CARRIAGE_RETURN = "\\r";
    public static final String ESCAPED_TABULATOR = "\\t";
    public static final String ESCAPED_NEW_LINE = "\\n";

    public static boolean isWhiteChar(char character) {
        switch (character){
            case NEW_LINE:
            case TABULATOR:
            case CARRIAGE_RETURN:
            case WHITE_SPACE:
                return true;
            default:
                return false;
        }
    }

    public static TokenType detectWhiteCharType(char character) {
        switch (character){
            case WHITE_SPACE:
                return TokenType.WHITE_SPACE;
            case TABULATOR:
                return TokenType.TABULATOR;
            case CARRIAGE_RETURN:
                return TokenType.CARRIAGE_RETURN;
            case NEW_LINE:
                return TokenType.NEW_LINE;
            default:
                return TokenType.UNKNOWN;
        }
    }

    public static String escapeWhiteChars(char character) {
        switch (character){
            case NEW_LINE:
                return ESCAPED_NEW_LINE;
            case TABULATOR:
                return ESCAPED_TABULATOR;
            case CARRIAGE_RETURN:
                return ESCAPED_CARRIAGE_RETURN;
            case WHITE_SPACE:
                return " ";
        }

        return "";
    }
}
